### ISSUE TITLE
PSA Kaggle competition ($60k)

### ISSUE THREAD

**Issue by gdevos010** [unit8co/darts#1506]
I want to let the community know there is a forecasting Kaggle competition, [GoDaddy - Microbusiness Density Forecasting](https://www.kaggle.com/competitions/godaddy-microbusiness-density-forecasting/overview). Usually, Kaggle time-series competitions are regression focused so it's cool that Darts can be used for this. It could be nice to have a starter notebook for the community. 

Please feel free to close this issue. I didn't know the best way to tell the community.

**Comment by hrzn** [comment]
Cool thanks for letting us know @gdevos010. Let's keep this issue around in case anyone wants to do something around that!

**Comment by SireeshLimbu** [comment]
Hi! Glad I found this comment. I was experimenting with darts for this competition, but I am stuck with a basic BlockRNN implementation. I have two questions, one specific and one broad, in that order :)

1. I encountered the same issue as mentioned [here](https://github.com/unit8co/darts/issues/299), regarding past covariates and  so I used TimeSeries.from_dataframe(df, time_col, value_cols) as @hrzn mentioned in that thread. I still get the same error 'The provided sequence of target series must have the same length as the provided sequence of covariate series'. Following is my simple LSTM implementation:

`# Where 'microbusiness_density' is the target
# df1 is the main dataframe after feature extraction and cleaning, it has 21 columns/covariates. All are 39 months long

X = df1.drop(['microbusiness_density'],axis = 1)
y =  df1[['microbusiness_density']]

from darts import TimeSeries
#df1.Index is of DataTime type 
X = TimeSeries.from_group_dataframe(X,group_cols = cols ,freq = '1MS')
y = TimeSeries.from_dataframe(y,freq = '1MS',value_cols = 'microbusiness_density')

from darts.models import BlockRNNModel
from darts import concatenate

model = BlockRNNModel(input_chunk_length=38,model = 'LSTM',dropout=0.2,random_state = 42, output_chunk_length = 1)

#Error here
model.fit(y, past_covariates=X)
`
The above code is only for one county. To summarize the competition problem statement, there are 3135 counties, each county has 39 months of multiple data columns/covariates (including microbusiness density). We have to forecast 6 months ahead for private leader board, 1 month for the public leaderboard. This brings me to the 2nd problem statement.

2. Is there a way to stack all the 3135 counties together, train and then forecast using DARTS? I am not asking for code, just a train of thought/ideas would be enough.

TIA!!

**Comment by madtoinou** [comment]
Challenge finished 9 days ago, top 3 solutions used LinearRegression, XGBRegressor and GRU respecively with a lot of interesting work on the data processing & covariates.
